sdk/cosmos/azure-spring-data-cosmos-test/src/test/java/com/azure/spring/data/cosmos/repository/integration/AddressRepositoryIT.java

```diff
@@ -56,24 +52,9 @@
 
     @Before
     public void setUp() {
-        if (!isSetupDone) {
-            staticTemplate = template;
-            template.createContainerIfNotExists(entityInformation);
-        }
-        repository.save(TEST_ADDRESS1_PARTITION1);
-        repository.saveAll(Lists.newArrayList(TEST_ADDRESS1_PARTITION2,
+        collectionManager.ensureContainersCreatedAndEmpty(template, Address.class);
+        repository.saveAll(Lists.newArrayList(TEST_ADDRESS1_PARTITION1, TEST_ADDRESS1_PARTITION2,
             TEST_ADDRESS2_PARTITION1, TEST_ADDRESS4_PARTITION3));
-        isSetupDone = true;
-    }
-
-    @After
-    public void cleanup() {
-        repository.deleteAll();
-    }
-
-    @AfterClass
-    public static void afterClassCleanup() {
-        staticTemplate.deleteContainer(entityInformation.getContainerName());
```

Comment:
Since we have removed these two APIs, `@After` and `@AfterClass` - who will handle the data and container delete operation for this integration test?

Reply:
`ensureContainersCreated`, called in the `@Before` ensures the container is created and all data within the container is deleted.  
The collectionManager is annotated with `@ClassRule`
```
    @ClassRule
    public static final IntegrationTestCollectionManager collectionManager = new IntegrationTestCollectionManager();
```
and the internal implementation will ensure the collections are deleted when the test is done (what used to be done in the `@AfterClass` method.

So, `@After` is not needed because the data is deleted in the @Before (since `ensureContainersCreated` is also responsible for creating the containers, you're guaranteed it must be invoked prior to inserting anything into the container) and `@AfterClass` is not needed because that is covered in the `@ClassRule`

Reply:
Makes sense!